source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,22 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char *uri) {
+  if (uriPattern == uri) {
+    return true;
+  }
+
+  size_t pattern_len = uriPattern.length();
+  if (pattern_len > 1 && uriPattern[pattern_len - 1] == '*' && uriPattern[pattern_len - 2] == '/') {
+    size_t uri_len = strlen(uri);
```

Comment:
perf nit: If you are going to call strlen() you might as well just walk both strings at the same time char by char. Also, probably don't need to do the direct compare first and then do this (could just do direct compare in same code path probably) but that's not as big of a deal to me.